tempesta_fw/t/functional/helpers/deproxy.py

```diff
@@ -135,9 +144,43 @@ def from_stream(rfile, no_crlf=False):
             headers.add(name, value)
         return headers
 
+    def _as_dict_lower(self):
+        ret = {}
+        for hed, val in self.items():
+            ret.setdefault(hed.lower(), []).append(val)
+        return ret
+
+    def _has_good_date(self):
+        return len(self.headers.get('date', [])) == 1
+
     def __eq__(self, other):
-        h_self = set([(hed.lower(), val) for hed, val in self.items()])
-        h_other = set([(hed.lower(), val) for hed, val in other.items()])
+        h_self = self._as_dict_lower()
+        h_other = other._as_dict_lower()
+
+        if self.is_expected and other.is_expected:
+            print("HeaderCollection: comparing two is_expected=True\n" +
```

Comment:
Because there are selftests on HeaderCollection that do not call `set_expected()`. I did not figure out how to solve this cleanly, so I'll just remove the check altogether. I could monkey-patch HeaderCollection inside selftests, but it is not re-imported...